server/src/main/java/com/google/coffeehouse/storagehandler/StorageHandler.java

```diff
@@ -0,0 +1,304 @@
+  * @param  membershipStatus  the enum specifying whether the user is a member or not
+  * @return clubs             the list of {@link Club}s
+  */
+  public static List<Club> getListOfClubs(DatabaseClient dbClient, String userId, MembershipConstants.MembershipStatus membershipStatus) {
+    Statement statement;
+    List<Club> clubs = new ArrayList<>();
+    if (membershipStatus == MembershipConstants.MembershipStatus.MEMBER) {
+      statement = 
+        Statement.newBuilder(
+                "SELECT userId, clubId "
+                  + "FROM Memberships "
+                  + "WHERE userId = @userId")
+            .bind("userId")
+            .to(userId)
+            .build();
+    } else {
+      statement = 
+        Statement.newBuilder(
+                "SELECT clubId "
+                  + "FROM Clubs "
+                  + "WHERE NOT EXISTS ("
+                  + "SELECT userId, clubId "
+                  + "FROM Memberships "
+                  + "WHERE userId = @userId)")
+            .bind("userId")
+            .to(userId)
+            .build();
+    }
+    try (ResultSet resultSet = dbClient.singleUse().executeQuery(statement)) {
+      while (resultSet.next()) {
+        String clubId = resultSet.getString("clubId");
+        clubs.add(getClub(dbClient, clubId));
+      }
+    }
+    return clubs;
+  }
+
+  /**
+  * Performs a transaction that adds a membership to the database.
+  * This method checks if a person is already a member of a club. If the person is
+  * already a member, it will throw an exception. Otherwise, it adds the row containing the
+  * user ID and club ID to the Memberships table.
+  *
+  * @param  dbClient    the database client
+  */
+  public static void getAddMembershipTransaction(DatabaseClient dbClient, String userId, String clubId) {
+    dbClient
+        .readWriteTransaction()
+        .run(
+          new TransactionCallable<Void>() {
+            @Override
+            public Void run(TransactionContext transaction) throws Exception {
+              Boolean exists = StorageHandlerHelper.checkMembership(transaction, userId, clubId);
+              if (!exists) {
+                transaction.buffer(StorageHandlerCommonMutations.addMembershipMutation(userId, clubId));
+              } else {
+                throw new IllegalArgumentException(MembershipConstants.PERSON_ALREADY_IN_CLUB);
+              }
+              return null;
+            }
+          }
+        );
+  }
+
+  /**
+  * Gets a transaction that deletes a membership from the database.
+  * This method checks if a person is already a member of a club by calling a read row transaction function
+  * from the StorageHandlerHelper class.
+  *
+  * @param  userId      the user ID string used to query the membership table
+  * @param  clubId      the club ID string used to query the membership table
+  */
+  public static void getDeleteMembershipTransaction(DatabaseClient dbClient, String userId, String clubId) {
```

Comment:
Sorry about the confusing wording. I changed the method names so that it is clearer, and I will be updating the Javadoc comments. They run/perform the transaction, so I changed it to `runAddMembershipTransacation()`. In the SotrageHandlerApi.java file, I define the wrapper which calls the methods here to just be `addMembership()`.